### DISCUSSION TITLE
Any plans to use psycopg (version 3) as an async db interface instead of aiopg?

### DISCUSSION THREAD

**Discussion by markc-versa** [procrastinate-org/procrastinate#671] [Ideas]
psycopg now has some async interfaces:

https://www.psycopg.org/psycopg3/docs/advanced/async.html

Is that something that could replace usage of aiopg as a connector?

**Comment by ewjoachim** [comment]
Definitely. Though I'm not spending a lot of time actively adding features to the project these days, but it's been in the back of my mind since v3 was released.

That said, I'm aware that connectors (especially sync vs async) in Procrastinate are already a confusing matter, and I'm missing some good dicussion on how to keep it maitnainable while avoiding making it more confusing.

Also, there's an ongoing PR on Asyncpg too #669
